### DISCUSSION TITLE
Using Cantaloupe with Keycloak and Mirador

### DISCUSSION THREAD

**Discussion by hamza-anaphore** [cantaloupe-project/cantaloupe#576] [General]
Hi, 

i have a question about how i can use [Cantaloupe ](https://cantaloupe-project.github.io/ ) with [Keycloak ](https://www.keycloak.org/documentation.html) and [Mirador ](https://github.com/ProjectMirador/mirador/), the user click the login button in Mirador after a Keycloak popup window will displayed, this window contains the login form, after that the user will authenticate so Keycloak will send the user informations to Mirador and Mirador will send those informations to Cantaloupe, after that Cantaloupe send the token to Keycloak and Keyclaok response to Cantaloupe with a response, so what i need to know is how the process with those three work ? i need some advices of the implementation 

Thanks in advance

**Comment by hrvoj3e** [comment]
I have some hints but not a solution
 - use delegate script in cantaloupe
 - see IIIF Auth specification

We will need a similar workflow but we are not there yet.
Btw. if you solved this please share it here.

**Comment by martin-hawk** [comment]
Hi,

I have the same issue. Does anyone found a solution?
